lib/Transforms/ConvertToCiphertextSemantics/ConvertToCiphertextSemantics.cpp

```diff
@@ -1231,6 +1239,353 @@ class ConvertTensorInsert
   }
 };
 
+// Attempt to infer statically the index of insertion. This case is used to
+// provide a simpler implementation of the insert/extract kernels when possible.
+static FailureOr<SmallVector<int64_t>> tryGetStaticIndices(
+    ArrayRef<Value> indices) {
+  SmallVector<int64_t> staticIndices;
+  for (Value v : indices) {
+    OpFoldResult staticIndex = getAsOpFoldResult(v);
+    if (auto attr = dyn_cast<Attribute>(staticIndex)) {
+      staticIndices.push_back(cast<IntegerAttr>(attr).getInt());
+    } else {
+      return failure();
+    }
+  }
+  return staticIndices;
```

Comment:
you can also use `getConstantIntValues(getAsOpFoldResult(indices)`

Reply:
Great call!